src/__test__/components/data-management/LaunchAnalysisButton.test.jsx

```diff
@@ -0,0 +1,322 @@
+const sample2Name = 'Sample 2';
+const sample2Uuid = 'sample-2';
+const experiment1id = 'experiment-1';
+
+const noDataState = {
+  projects: {
+    ...initialProjectState,
+    meta: {
+      ...initialProjectState.meta,
+      activeProjectUuid: projectUuid,
+      loading: false,
+    },
+    ids: [projectUuid],
+    [projectUuid]: {
+      ...projectTemplate,
+      experiments: [experiment1id],
+      uuid: projectUuid,
+      name: projectName,
+      description: projectDescription,
+    },
+  },
+  experiments: {
+    ...initialExperimentsState,
+    ids: [experiment1id],
+    [experiment1id]: {
+      ...experimentTemplate,
+    },
+  },
+  samples: {
+    ...initialSamplesState,
+  },
+  backendStatus: {
+    [experiment1id]: {
+      ...initialExperimentBackendStatus,
+      status: {
+        gem2s: {
+          status: PipelineStatus.NOT_CREATED,
+        },
+        pipeline: {
+          status: PipelineStatus.NOT_CREATED,
+        },
+      },
+    },
+  },
+};
+
+const withDataState = {
+  ...noDataState,
+  projects: {
+    ...noDataState.projects,
+    [projectUuid]: {
+      ...noDataState.projects[projectUuid],
+      samples: [sample1Uuid, sample2Uuid],
+      metadataKeys: ['metadata-1'],
+    },
+  },
+  experiments: {
+    ...noDataState.experiments,
+    [experiment1id]: {
+      ...experimentTemplate,
+      ...noDataState.experiments[experiment1id],
+      sampleIds: [sample1Uuid, sample2Uuid],
+    },
+  },
+  samples: {
+    ...noDataState.samples,
+    [sample1Uuid]: {
+      ...sampleTemplate,
+      name: sample1Name,
+      projectUuid,
+      uuid: sample1Uuid,
+      type: '10X Chromium',
+      metadata: ['value-1'],
+      fileNames: ['features.tsv.gz', 'barcodes.tsv.gz', 'matrix.mtx.gz'],
+      files: {
+        'features.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'barcodes.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'matrix.mtx.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+      },
+    },
+    [sample2Uuid]: {
+      ...sampleTemplate,
+      name: sample2Name,
+      projectUuid,
+      uuid: sample2Uuid,
+      type: '10X Chromium',
+      metadata: ['value-2'],
+      fileNames: ['features.tsv.gz', 'barcodes.tsv.gz', 'matrix.mtx.gz'],
+      files: {
+        'features.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'barcodes.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'matrix.mtx.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+      },
+    },
+  },
+  backendStatus: {
+    [experiment1id]: {
+      ...initialExperimentBackendStatus,
+      status: {
+        gem2s: {
+          paramsHash: 'old-params-hash',
+          status: PipelineStatus.SUCCEEDED,
+        },
+        pipeline: {
+          status: PipelineStatus.SUCCEEDED,
+        },
+      },
+    },
+  },
+};
+
+describe('LaunchAnalysisButton', () => {
```

Comment:
These look like some thorough tests Agi, they will be very handy!

Reply:
Thanks Iva!